### PR TITLE
move common tools from `nu-command` to `nu-cmd-base`

### PR DESCRIPTION
**Pull request by amtoine**
related to 
- https://github.com/nushell/nushell/pull/9404

# Description
to support our cratification effort and moving non-1.0 commands outside of the main focus, this PR
- creates a new `nu-cmd-base` crate to hold the common structs, traits and functions used by all command-related crates
- to start the transition, moves the `input_handler` module from `nu-command` to `nu-cmd-base`

# User-Facing Changes
```
$nothing
```

# Tests + Formatting
- :green_circle: `toolkit fmt`
- :green_circle: `toolkit clippy`
- :black_circle: `toolkit test`
- :black_circle: `toolkit test stdlib`

# After Submitting
```
$nothing
```

### CODE REVIEW

**Comment by amtoine** [comment]
cc/ @stormasm 

i've used the following command to isolate the tool items used throughout the `nu-command` crate:
```bash
rg "^use crate" crates/nu-command/src/math/ | lines | parse "{file}:{match}" | get match
```
which gives, after removing the `use crate` from the _default_context_ and from the help command which obviously need these imports, we get the following table of items
```
╭────┬───────────────────────────────────────────────────────────────────────────────────────────────╮
│  1 │ use crate::math::reducers::{reducer_for, Reduce};                                             │
│  2 │ use crate::math::utils::run_with_function;                                                    │
│  3 │ use crate::input_handler::{operate, CmdArgument};                                             │
│  4 │ use crate::input_handler::{operate, CellPathOnlyArgs};                                        │
│  5 │ use crate::formats::to::delimited::merge_descriptors;                                         │
│  6 │ use crate::formats::nu_xml_format::{COLUMN_ATTRS_NAME, COLUMN_CONTENT_NAME, COLUMN_TAG_NAME}; │
│  7 │ use crate::formats::to::delimited::to_delimited_data;                                         │
│  8 │ use crate::math::avg::average;                                                                │
│ 13 │ use crate::input_handler::{operate as general_operate, CmdArgument};                          │
│ 14 │ use crate::util::get_guaranteed_cwd;                                                          │
│ 15 │ use crate::date::utils::parse_date_from_string;                                               │
│ 16 │ use crate::{generate_strftime_list, parse_date_from_string};                                  │
│ 17 │ use crate::formats::value_to_string;                                                          │
│ 18 │ use crate::help::highlight_search_string;                                                     │
│ 19 │ use crate::hook::eval_hook;                                                                   │
│ 20 │ use crate::database::values::definitions::{db_row::DbRow, db_table::DbTable};                 │
│ 21 │ use crate::database::values::sqlite::open_sqlite_db;                                          │
│ 22 │ use crate::ExternalCommand;                                                                   │
│ 23 │ use crate::network::http::client::{                                                           │
│ 24 │ use crate::formats::value_to_json_value;                                                      │
│ 25 │ use crate::database::values::definitions::db_row::DbRow;                                      │
│ 26 │ use crate::filesystem::util::FileStructure;                                                   │
│ 27 │ use crate::progress_bar;                                                                      │
│ 28 │ use crate::database::SQLiteDatabase;                                                          │
│ 29 │ use crate::DirBuilder;                                                                        │
│ 30 │ use crate::DirInfo;                                                                           │
│ 31 │ use crate::{DirBuilder, DirInfo, FileInfo};                                                   │
│ 32 │ use crate::filesystem::cd_query::query;                                                       │
│ 33 │ use crate::debug::inspect_table::{                                                            │
│ 34 │ use crate::{grapheme_flags, util};                                                            │
│ 35 │ use crate::grapheme_flags;                                                                    │
│ 36 │ use crate::operate;                                                                           │
╰────┴───────────────────────────────────────────────────────────────────────────────────────────────╯
```

## :question: do we move them all to the new `nu-cmd-base` crate? :open_mouth:

**Comment by stormasm** [comment]
@amtoine 

I think to be on the safe side of things we should move those helper functions over as we need them...

In other words the first one I would move in this PR is input_handler because the *bytes* command will need it...

So input_handler will be our first command in nu-cmd-base

and it will be up and "running" / landed

then I will do the bytes command

and then as we want to move more commands 

we move the helper commands first and then the actual commands...

What do you think of this ?

**Comment by amtoine** [comment]
sounds good to me :+1: 

so let's move `input_handler` :smirk:

File: crates/nu-command/Cargo.toml [hunk 1]
```diff
@@ -96,6 +96,7 @@ url = "2.2"
 uuid = { version = "1.3", features = ["v4"] }
 wax = { version = "0.5" }
 which = { version = "4.4", optional = true }
+nu-cmd-base = { version = "0.80.1", path = "../nu-cmd-base" }
```
Comment (sholderbach) [inline crates/nu-command/Cargo.toml:99]: Watch out: this is behind the main versions and would get missed by our regex version bump

Comment (amtoine) [reply crates/nu-command/Cargo.toml:99]: very good catch :+1: 
i think i did create this crate before the last release :open_mouth:

Comment (amtoine) [reply crates/nu-command/Cargo.toml]: it should be a lot better in 4de6bf7cd8c3565450ef23e689e1e5caf2a69214 :wink:

**Comment by stormasm** [comment]
@kubouch @fdncred you all can go ahead and land this PR if you think its ok...

or I will check in with you all tomorrow morning after I wake up and we can figure out what to do....

thanks !

**Comment by stormasm** [comment]
all this PR is doing is moving input_handler to a new crate called *nu-cmd-base*

**Comment by stormasm** [comment]
@amtoine I went ahead and merged this as it was blocking me making forward progress on moving the bytes command to nu-cmd-extra...

Now I should be able to land that PR as well....

**Comment by stormasm** [comment]
And now that we have seen some other PRs fail in the exact same spot with Python virtualenv I was assured the issue is NOT related to this PR 😄

**Pull request merged**